### PR TITLE
feat(Select): add flag to not reset on select, add event to toggle callback

### LINKED ISSUE #6641: Typeahead Multi Select: filter is reset after select
**Describe the issue. What is the expected and unexpected behavior?**
Unexpected behavior in the following user story:
As a user working with the `Select` component example as linked below
- I enter 'new' in the type ahead input text,
- I see the filtered result,
- I select the first option (New Jersey),
- My filter has been reset, so the filter input text is removed after the first selection
image

Expected behavior:

- My filter is not reset and I can select another record in the filtered result 
image

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

https://www.patternfly.org/v4/components/select/#multiple-with-custom-chip-group-props

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
enhancement

**What is your product and what release version are you targeting?**
"@patternfly/react-core": "^4.162.3"

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #6641 & #6835

### CODE REVIEW

File: packages/react-core/src/components/Select/Select.tsx [hunk 1]
```diff
@@ -159,6 +159,8 @@ export interface SelectProps
   isInputValuePersisted?: boolean;
   /** @beta Flag for retaining filter results on blur from keyboard-entered typeahead text */
   isInputFilterPersisted?: boolean;
+  /** Flag indicating that the state of Select such as typeahead input should reset upon selection */
```
Comment (nicolethoen) [inline packages/react-core/src/components/Select/Select.tsx:162]: is it the state or the value of the input? as is, this prop sounds very destructive 😳

**Review by nicolethoen — CHANGES_REQUESTED** [review]
This looks good - the only interaction concern/possible regression I could find:
Mar-21-2022 12-23-54

It's doing weird things with the description text like the gif above.

File: packages/react-core/src/components/Select/Select.tsx (hunk 1 above)
Comment (kmcfaul) [reply packages/react-core/src/components/Select/Select.tsx:162]: I'll update the description to remove the state mention as its internal.

Will also look into the description popping up.

**Review by nicolethoen — APPROVED** [review]

**Comment by tlabaj** [comment]
this would be a great candidate for a test to verify adding the flag works as expected.

**Review by wise-king-sullyman — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]
Looks great!

**Pull request merged**